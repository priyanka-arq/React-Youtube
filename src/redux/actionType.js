//login actions
export const LOGIN_SUCCESS = "LOGIN_SUCCESS";
export const LOGIN_FAILURE = "LOGIN_FAILURE";
export const LOGIN_REQUEST = "LOGIN_REQUEST";
export const LOGOUT = "LOGOUT";
export const LOAD_PROFILE = "LOAD_PROFILE";

//video actions
export const HOME_VIDEOS_SUCCESS = "HOME_VIDEOS_SUCCESS";
export const HOME_VIDEOS_FAILURE = "HOME_VIDEOS_FAILURE";
export const HOME_VIDEOS_REQUEST = "HOME_VIDEOS_REQUEST";
export const SET_VIDEO_DETAILS = "SET_VIDEO_DETAILS";
// // export const SET_CHANNEL_ICON = "SET_CHANNEL_ICON";

export const SELECTED_VIDEO_REQUEST = "SELECTED_VIDEO_REQUEST";
export const SELECTED_VIDEO_SUCCESS = "SELECTED_VIDEO_SUCCESS";
export const SELECTED_VIDEO_FAILURE = "SELECTED_VIDEO_FAILURE";

//channel actions
export const CHANNEL_DETAILS_REQUEST = "CHANNEL_DETAILS_REQUEST";
export const CHANNEL_DETAILS_SUCCESS = "CHANNEL_DETAILS_SUCCESS";
export const CHANNEL_DETAILS_FAILURE = "CHANNEL_DETAILS_FAILURE";

//subscription
export const SET_SUBSCRIPTION_STATUS = "SET_SUBSCRIPTION_STATUS";

//comments
export const COMMENTS_LIST_REQUEST = "COMMENTS_LIST_REQUEST";
export const COMMENTS_LIST_SUCCESS = "COMMENTS_LIST_SUCCESS";
export const COMMENTS_LIST_FAILURE = "COMMENTS_LIST_FAILURE";

//add comment
export const CREATE_COMMENT_SUCCESS = "CREATE_COMMENT_SUCCESS";
export const CREATE_COMMENT_FAILURE = "CREATE_COMMENT_FAILURE";

//related videos
export const RELATED_VIDEOS_REQUEST = "RELATED_VIDEOS_REQUEST";
export const RELATED_VIDEOS_SUCCESS = "RELATED_VIDEOS_SUCCESS";
export const RELATED_VIDEOS_FAILURE = "RELATED_VIDEOS_FAILURE";

//Search videos
export const SEARCH_VIDEO_REQUEST = "SEARCH_VIDEO_REQUEST";
export const SEARCH_VIDEO_SUCCESS = "SEARCH_VIDEO_SUCCESS";
export const SEARCH_VIDEO_FAILURE = "SEARCH_VIDEO_FAILURE";

//Subscriptions channel action
export const SUBSCRIPTIONS_CHANNEL_REQUEST = "SUBSCRIPTIONS_CHANNEL_REQUEST";
export const SUBSCRIPTIONS_CHANNEL_SUCCESS = "SUBSCRIPTIONS_CHANNEL_SUCCESS";
export const SUBSCRIPTIONS_CHANNEL_FAILURE = "SUBSCRIPTIONS_CHANNEL_FAILURE";

//channel videos actions
export const CHANNEL_VIDEOS_REQUEST = "CHANNEL_VIDEOS_REQUEST";
export const CHANNEL_VIDEOS_SUCCESS = "CHANNEL_VIDEOS_SUCCESS";
export const CHANNEL_VIDEOS_FAILURE = "CHANNEL_VIDEOS_FAILURE";

//liked videos action
export const LIKED_VIDEOS_REQUEST = "LIKED_VIDEOS_REQUEST";
export const LIKED_VIDEOS_SUCCESS = "LIKED_VIDEOS_SUCCESS";
export const LIKED_VIDEOS_FAILURE = "LIKED_VIDEOS_FAILURE";

//subscribe Channel
export const SUBSCRIBE_CHANNEL_REQUEST = "SUBSCRIBE_CHANNEL_REQUEST";
export const SUBSCRIBE_CHANNEL_SUCCESS = "SUBSCRIBE_CHANNEL_SUCCESS";
export const SUBSCRIBE_CHANNEL_FAILURE = "SUBSCRIBE_CHANNEL_FAILURE";

//subscribe Channel
export const LIKE_VIDEO_REQUEST = "LIKE_VIDEO_REQUEST";
export const LIKED_VIDEO_SUCCESS = "LIKED_VIDEO_SUCCESS";
export const LIKED_VIDEO_FAILURE = "LIKED_VIDEO_FAILURE";
