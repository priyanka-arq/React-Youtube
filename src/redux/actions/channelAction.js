import networkRequest from "../../api";
import {
  CHANNEL_DETAILS_FAILURE,
  CHANNEL_DETAILS_REQUEST,
  CHANNEL_DETAILS_SUCCESS,
} from "../actionType";

export const getChannelDetails = (id) => async (dispatch) => {
  dispatch({
    type: CHANNEL_DETAILS_REQUEST,
  });
  try {
    const response = await networkRequest.get("/channels", {
      params: {
        part: "snippet,statistics, contentDetails",
        id: id,
      },
    });

    dispatch({
      type: CHANNEL_DETAILS_SUCCESS,
      payload: response.data.items[0],
    });
  } catch (error) {
    dispatch({
      type: CHANNEL_DETAILS_FAILURE,
      payload: error.message,
    });
  }
};
