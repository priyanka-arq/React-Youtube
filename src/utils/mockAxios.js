//############################//

//inside src => craete __mocks__ => axios.js
//paste following code

const mockAxios = jest.genMockFromModule("axios");

// this is the key to fix the axios.create() undefined error!
mockAxios.create = jest.fn(() => mockAxios);
mockAxios.post = jest.fn(() => mockAxios);
mockAxios.get = jest.fn(() => mockAxios);

export default mockAxios;

// export default {
//   get: jest.fn(() => Promise.resolve({ data: {} })),
//   create: jest.fn(() => Promise.resolve({ data: {} })),
// };

//#########################################

import mockAxios from "axios";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";

import {
  getCommentsDetails,
  addComment,
} from "../redux/actions/commentListAction";

import {
  COMMENTS_LIST_REQUEST,
  COMMENTS_LIST_SUCCESS,
  CREATE_COMMENT_SUCCESS,
  CREATE_COMMENT_FAILURE,
  COMMENTS_LIST_FAILURE,
} from "../redux/actionType";

let actions;
let store;
const createMockStore = configureMockStore([thunk]);

describe("get comments list action", () => {
  it("it returns data when api is called", async function () {
    const data = { response: true };
    const comments = [
      {
        kind: "youtube#commentThread",
        etag: "9SadOUrGkPQSt0sorsFijKTBL80",
        id: "Ugwybolefx_P4E6phah4AaABAg",
        snippet: {},
      },
      {
        kind: "youtube#commentThread",
        etag: "9SadOUrGkPQSt0sorsFijKTBL80",
        id: "Ugwybolefx_P4E6phah4AaABAg",
        snippet: {},
      },
    ];
    //mock.onGet("/commentThreads").reply(200, comments);

    mockAxios.get.mockImplementationOnce(() =>
      Promise.resolve({
        data: { results: comments },
      })
    );

    // expect(mockAxios.get).toHaveBeenCalledTimes(1);

    store = createMockStore({
      commentList: [
        {
          kind: "youtube#commentThread",
          etag: "9SadOUrGkPQSt0sorsFijKTBL80",
          id: "Ugwybolefx_P4E6phah4AaABAg",
          snippet: {},
        },
        {
          kind: "youtube#commentThread",
          etag: "9SadOUrGkPQSt0sorsFijKTBL80",
          id: "Ugwybolefx_P4E6phah4AaABAg",
          snippet: {},
        },
      ],
    });

    store.dispatch(getCommentsDetails("abc")).then(() => {
      expect(store.getActions()).toContainEqual({
        type: COMMENTS_LIST_REQUEST,
      });
    });

    //expect(mock.history.get.length).toBe(1);
  });
});

///###########################
describe("add comment", () => {
  const store = createMockStore({
    auth: {
      accessToken: "AIzaSyCLg4cHF4d0Sj_UwQQhV1JPvCngteYtNkM",
    },
  });

  test("add comment action", () => {
    const comment = "nice";
    const auth = { accessToken: "AIzaSyCLg4cHF4d0Sj_UwQQhV1JPvCngteYtNkM" };

    mockAxios.post("/commentThreads", comment, {
      params: {
        part: "snippet ",
      },
      headers: {
        Authorization: `Bearer ${auth.accessToken}`,
      },
    });

    expect(mockAxios.post).toHaveBeenCalledTimes(1);

    store.dispatch(addComment("abc", "xyz")).then(() => {
      actions = store.getActions();
      expect(store.getActions()).toMatchSnapshot();

      expect(actions[0]).toEqual({
        type: CREATE_COMMENT_SUCCESS,
      });
    });
  });
});

describe("commentList action", () => {
  const store = createMockStore({});

  it("gets a comments list", () =>
    store.dispatch(getCommentsDetails("abc")).then(() =>
      expect(store.getActions()).toContainEqual({
        type: COMMENTS_LIST_FAILURE,
      })
    ));
});

describe("fetches comments ", async () => {
  const comments = [
    {
      kind: "youtube#commentThread",
      etag: "9SadOUrGkPQSt0sorsFijKTBL80",
      id: "Ugwybolefx_P4E6phah4AaABAg",
      snippet: {},
    },
    {
      kind: "youtube#commentThread",
      etag: "9SadOUrGkPQSt0sorsFijKTBL80",
      id: "Ugwybolefx_P4E6phah4AaABAg",
      snippet: {},
    },
  ];

  it("should execute fetch data", () => {
    store = createMockStore();
    // Return the promise
    return store.dispatch(getCommentsDetails()).then(() => {
      const actions = store.getActions();
      expect(actions[0]).toEqual({ type: COMMENTS_LIST_REQUEST });
    });
  });
});
