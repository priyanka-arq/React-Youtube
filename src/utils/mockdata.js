import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";

const mockStore = configureMockStore([thunk]);

export const store = mockStore({
  auth: {
    accessToken:
      "ya29.A0ARrdaM9KG5nxzuU4bOhVbHKu2EDj2lKG36NY7wtm8IVsKLEbo3_O699EDk8IxvhccuIHPUyhQTT5pXxmCAavDL4iTI7iblxFuIXDONABwlhi8LQHbhvxGfMn27FnCksi2qJhg4hn7-IzIcHxildJ0RIn2ragrw",
    user: {
      name: "Priyanka Patel",
      photoURL:
        "https://lh3.googleusercontent.com/a-/AOh14GgEX3SB8FVF-3OLqmdT3zAW0vRL6alhlseVN0bn=s96-c",
    },
    error: null,
    loading: false,
  },
  commentList: [
    {
      kind: "youtube#commentThread",
      etag: "9SadOUrGkPQSt0sorsFijKTBL80",
      id: "Ugwybolefx_P4E6phah4AaABAg",
      snippet: {},
    },
    {
      kind: "youtube#commentThread",
      etag: "9SadOUrGkPQSt0sorsFijKTBL80",
      id: "Ugwybolefx_P4E6phah4AaABAg",
      snippet: {},
    },
  ],
  homeVideos: {
    videos: [
      {
        kind: "youtube#video",
        etag: "8XDUSTuq-9vHEgBH-g3R1QOmD30",
        id: "ATElufr0OiE",
        snippet: {
          publishedAt: "2022-01-31T16:27:42Z",
          channelId: "UCTNtRdBAiZtHP9w7JinzfUg",
          title:
            "Mudhal Nee Mudivum Nee - Title Track Video | Darbuka Siva | Sid Sriram | Thamarai",
          description:
            "A lilting melody from the famed #DarbukaSiva , #SidSriram and #Thamarai trio! Presenting the loop-worthy #MudhalNeeMudivumNee title track from the composer's directorial debut!\n\nHit play and enjoy the official Tamil song video of #MNMN now!\n\nStream now - https://SMI.lnk.to/MNMNTitleTrackDC\n\nComposer's Note:\n\nThe song Mudhal Nee Mudivum Nee for me is a collage of sounds from various musical soundscapes that relate to nostalgia and the wonderful memories associated with it. \n\nThe Sarodâ€™s classical texture converses with the voice to create that feeling, coupled with strings that form a sonic movement which often plays around with trancy layers of interesting shapes and patterns of sounds.\n\nDarbuka Siva\n\nMovie - Mudhal Nee Mudivum Nee\nSong - Mudhal Nee Mudivum Nee\nSinger - Sid Sriram, Darbuka Siva\nComposer - Darbuka Siva\nLyrics - Thamarai\nWritten and Directed by Darbuka Siva\nDOP - Sujith Saarang\nEditing - Sreejith Saarang\nProduced by Super Talkies - Sameer Bharat Ram\n\nMusicians:\nSarod: Debasmita Bhattacharya\nElectric Guitar, Acoustic guitar, Banjo, Ukulele: Keba Jeremiah\nKeyboard: Shyam Benjamin \nMusic arranged and programmed by Darbuka Siva\nMixed and Mastered by Balu Thankachan at 20dB Soundstudios, Chennai.\nAssisted by Elwin Joseph, Hariharan\n \nRecorded at \nSeed Studios, Chennai\n20 dB Soundstudios, Chennai\nRJ Studio, Kolkata\nRecording Engineers: Dheeleben, Hafeez, Avinash Sathish, Rana Mondal\n \nLyric Video Credits: \nArtwork Hand drawn by Purva Raghunath\nVideo Effects: Pradeep\nVideo Editing: Darbuka Siva\n\nMusic Label - Sony Music Entertainment India Pvt. Ltd.\n\nÂ© 2022 Sony Music Entertainment India Pvt. Ltd.\n\nSubscribe Now: http://bit.ly/SonyMusicSouthVevo\nSubscribe Now: http://bit.ly/SonyMusicSouthYT\nFollow us: https://www.instagram.com/sonymusic_south/\nFollow us: Twitter: https://twitter.com/SonyMusicSouth\nLike us: Facebook: https://www.facebook.com/SonyMusicSouth\n\nhttp://vevo.ly/TSdD3a",
          thumbnails: {
            default: {
              url: "https://i.ytimg.com/vi/ATElufr0OiE/default.jpg",
              width: 120,
              height: 90,
            },
            medium: {
              url: "https://i.ytimg.com/vi/ATElufr0OiE/mqdefault.jpg",
              width: 320,
              height: 180,
            },
            high: {
              url: "https://i.ytimg.com/vi/ATElufr0OiE/hqdefault.jpg",
              width: 480,
              height: 360,
            },
            standard: {
              url: "https://i.ytimg.com/vi/ATElufr0OiE/sddefault.jpg",
              width: 640,
              height: 480,
            },
            maxres: {
              url: "https://i.ytimg.com/vi/ATElufr0OiE/maxresdefault.jpg",
              width: 1280,
              height: 720,
            },
          },
          channelTitle: "SonyMusicSouthVEVO",
          tags: [
            "Sony Music South",
            "Sony Music",
            "Latest Song",
            "Tamil song",
            "Tamil Latest Song",
            "Tamil Latest Songs 2018",
            "Tamil Latest Songs 2019",
            "Latest songs in Tamil",
            "Songs in Tamil",
            "Latest Tamil Movies",
            "Tamil Movie Songs",
            "Latest Video Songs Tamil",
            "latest Tamil Love Songs",
            "Latest Tamil Hit Songs",
            "darbuka siva songs",
            "sid sriram songs",
            "sid sriram tamil hits",
            "darbuka siva tamil songs",
            "mudhalum nee mudivum nee trailer",
            "Darbuka Siva",
            "Mudhal Nee Mudivum Nee",
            "Sony",
            "Pvt.",
          ],
          categoryId: "10",
          liveBroadcastContent: "none",
          localized: {
            title:
              "Mudhal Nee Mudivum Nee - Title Track Video | Darbuka Siva | Sid Sriram | Thamarai",
            description:
              "A lilting melody from the famed #DarbukaSiva , #SidSriram and #Thamarai trio! Presenting the loop-worthy #MudhalNeeMudivumNee title track from the composer's directorial debut!\n\nHit play and enjoy the official Tamil song video of #MNMN now!\n\nStream now - https://SMI.lnk.to/MNMNTitleTrackDC\n\nComposer's Note:\n\nThe song Mudhal Nee Mudivum Nee for me is a collage of sounds from various musical soundscapes that relate to nostalgia and the wonderful memories associated with it. \n\nThe Sarodâ€™s classical texture converses with the voice to create that feeling, coupled with strings that form a sonic movement which often plays around with trancy layers of interesting shapes and patterns of sounds.\n\nDarbuka Siva\n\nMovie - Mudhal Nee Mudivum Nee\nSong - Mudhal Nee Mudivum Nee\nSinger - Sid Sriram, Darbuka Siva\nComposer - Darbuka Siva\nLyrics - Thamarai\nWritten and Directed by Darbuka Siva\nDOP - Sujith Saarang\nEditing - Sreejith Saarang\nProduced by Super Talkies - Sameer Bharat Ram\n\nMusicians:\nSarod: Debasmita Bhattacharya\nElectric Guitar, Acoustic guitar, Banjo, Ukulele: Keba Jeremiah\nKeyboard: Shyam Benjamin \nMusic arranged and programmed by Darbuka Siva\nMixed and Mastered by Balu Thankachan at 20dB Soundstudios, Chennai.\nAssisted by Elwin Joseph, Hariharan\n \nRecorded at \nSeed Studios, Chennai\n20 dB Soundstudios, Chennai\nRJ Studio, Kolkata\nRecording Engineers: Dheeleben, Hafeez, Avinash Sathish, Rana Mondal\n \nLyric Video Credits: \nArtwork Hand drawn by Purva Raghunath\nVideo Effects: Pradeep\nVideo Editing: Darbuka Siva\n\nMusic Label - Sony Music Entertainment India Pvt. Ltd.\n\nÂ© 2022 Sony Music Entertainment India Pvt. Ltd.\n\nSubscribe Now: http://bit.ly/SonyMusicSouthVevo\nSubscribe Now: http://bit.ly/SonyMusicSouthYT\nFollow us: https://www.instagram.com/sonymusic_south/\nFollow us: Twitter: https://twitter.com/SonyMusicSouth\nLike us: Facebook: https://www.facebook.com/SonyMusicSouth\n\nhttp://vevo.ly/TSdD3a",
          },
          defaultAudioLanguage: "en",
        },
        contentDetails: {
          duration: "PT5M36S",
          dimension: "2d",
          definition: "hd",
          caption: "false",
          licensedContent: true,
          regionRestriction: {
            blocked: ["US"],
          },
          contentRating: {},
          projection: "rectangular",
        },
        statistics: {
          viewCount: "2088681",
          likeCount: "133120",
          favoriteCount: "0",
          commentCount: "2850",
        },
      },
      {
        kind: "youtube#video",
        etag: "PK0np5HEbZ3Cz6FX3hZ7aSXu4As",
        id: "67Fuq_oRPMw",
        snippet: {
          publishedAt: "2022-01-31T03:42:51Z",
          channelId: "UC1J-EgBVeo-TUaC5m0t6bKg",
          title: "balloon and syringe experiment ðŸ¤© gone wrong ðŸ¤¯ #shorts",
          description: "",
          thumbnails: {
            default: {
              url: "https://i.ytimg.com/vi/67Fuq_oRPMw/default.jpg",
              width: 120,
              height: 90,
            },
            medium: {
              url: "https://i.ytimg.com/vi/67Fuq_oRPMw/mqdefault.jpg",
              width: 320,
              height: 180,
            },
            high: {
              url: "https://i.ytimg.com/vi/67Fuq_oRPMw/hqdefault.jpg",
              width: 480,
              height: 360,
            },
          },
          channelTitle: "Mr. Easy Tamil",
          tags: [
            "craft tamil channel",
            "craft tamil",
            "experiment in tamil",
            "experiment tamil",
            "easy experiment",
            "science experiment",
            "magic tricks",
            "balloon experiment gone wrong",
            "Mr easy tamil",
            "laminar flow experiment",
            "satisfying experiment",
          ],
          categoryId: "24",
          liveBroadcastContent: "none",
          defaultLanguage: "en-IN",
          localized: {
            title: "balloon and syringe experiment ðŸ¤© gone wrong ðŸ¤¯ #shorts",
            description: "",
          },
          defaultAudioLanguage: "ta",
        },
        contentDetails: {
          duration: "PT1M",
          dimension: "2d",
          definition: "hd",
          caption: "false",
          licensedContent: true,
          contentRating: {},
          projection: "rectangular",
        },
        statistics: {
          viewCount: "2531259",
          likeCount: "158897",
          favoriteCount: "0",
          commentCount: "66",
        },
      },
    ],
    activeCategory: "react",
    loading: false,
  },
  selectedVideos: { loading: false, videos: {}, error: null },
  channelDetails: { loading: false, channel: {}, error: null },
  subscriptionStatus: { subscriptionStatus: false },
  relatedVideos: { loading: false, videos: [], error: null },
  searchVideos: { loading: false, videos: [], error: null },
  subscriptionChannelVideos: { loading: false, videos: [], error: null },
  channelVideos: { loading: false, videos: [], error: null },
  likedVideos: { loading: false, videos: [], error: null },
});
