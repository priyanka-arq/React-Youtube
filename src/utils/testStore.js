import { createStore, applyMiddleware, combineReducers } from "redux";
import { rootReducer, middlewares } from "../redux/store";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

export const findByTestAtrr = (component, attr) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};
export const testStore = (initialState = {}) => {
  const createStoreWithMiddleware = applyMiddleware(...middlewares)(
    createStore
  );

  return createStoreWithMiddleware(rootReducer, initialState);
};
// const mockStore = configureStore([thunk]);

// export const testStore = (state = {}) => {
//   return mockStore({
//     ...state,
//   });
// };
