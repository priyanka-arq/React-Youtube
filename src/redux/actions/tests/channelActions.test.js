import { store } from "../../../utils/mockdata";
import { getChannelDetails } from "../channelAction";
import { CHANNEL_DETAILS_REQUEST } from "../../actionType";
//import moxios from 'moxios'

describe("CHANNEL_DETAILS ACTION", () => {
  test("dispatch channel details actions", () => {
    store.dispatch(getChannelDetails()).then(() => {
      const actions = store.getActions();
      expect(store.getActions()).toMatchSnapshot();
      console.log(actions);
      expect(actions[0]).toEqual({ type: CHANNEL_DETAILS_REQUEST });
    });
    expect(store.getActions()).toMatchSnapshot();
  });
});

//#########    using moxios   ###########
// const mockSuccess = (data) => ({ status: 200, response: { data } });
// const mockError = (data) => ({ status: 500, response: { error } });

// describe("Fetch channel details", () => {
//   beforeEach(() => moxios.install());
//   afterEach(() => moxios.uninstall());

//   test("API call and get channel details", () => {
//     const channel = {
//       kind: "youtube#channle",
//       id: "123",
//       etag: "xDjIUZmbJ8efm6xXRpUps1-9Q6k",
//       snippet: {},
//       contentDetails: {},
//       statistics: {},
//     };

//     const error = "";

//     moxios.wait(() => {
//       const request = moxios.requests.mostRecent();
//       request.respondWith(mockSuccess(channel));
//     });

//     const expected = [
//       { type: CHANNEL_DETAILS_REQUEST },
//       {
//         type: CHANNEL_DETAILS_SUCCESS,
//         payload: channel,
//       },
//       {
//         type: CHANNEL_DETAILS_FAILURE,
//         payload: error,
//       },
//     ];
//     const channelDetails = getChannelDetails("123");
//     store.dispatch(getChannelDetails("123")).then(() => {
//       const actions = store.getActions();
//       expect(actions[0]).toEqual({
//         type: CHANNEL_DETAILS_REQUEST,
//       });
//       expect(actions[1]).toEqual(expected[1]);
//       expect(actions[0]).toEqual(getChannelDetails());
//       expect(channelDetails).toEqual({
//         kind: "youtube#channle",
//         id: "123",
//         etag: "xDjIUZmbJ8efm6xXRpUps1-9Q6k",
//         snippet: {},
//         contentDetails: {},
//         statistics: {},
//       });
//     });
//   });
// });
