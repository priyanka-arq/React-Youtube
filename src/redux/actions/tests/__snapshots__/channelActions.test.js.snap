// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CHANNEL_DETAILS ACTION dispatch channel details actions 1`] = `
Array [
  Object {
    "type": "CHANNEL_DETAILS_REQUEST",
  },
]
`;
